src/modules/Session/model.ts

```diff
@@ -0,0 +1,108 @@
+import { inputObjectType, objectType } from 'nexus/dist';
+import { PrismaClient, Prisma } from '@prisma/client';
+import { randomUUID } from 'crypto';
+import { ApolloError } from 'apollo-server-core';
+import { token } from '../../helpers';
+
+export const Session = objectType({
+  name: 'Session',
+  definition(t) {
+    t.nonNull.string('id');
+    t.nonNull.date('createdAt');
+    t.nonNull.string('createdBy');
+    t.date('referenceExpiryDate');
+    t.nonNull.string('referenceTokenId');
+    t.nonNull.boolean('isUserCreated');
+    t.string('name');
+    t.date('revokedAt');
+  },
+});
+
+export const SessionCreate = inputObjectType({
+  name: 'SessionCreate',
+  definition(t) {
+    t.date('referenceExpiryDate');
+    t.nonNull.string('name');
+  },
+});
+
+export const SessionCreateOutput = objectType({
+  name: 'SessionCreateOutput',
+  definition(t) {
+    t.nonNull.field('session', { type: 'Session' });
+    t.nonNull.string('token');
+  },
+});
+
+async function newSession(
+  prisma: PrismaClient,
+  session: Prisma.SessionCreateInput,
+) {
+  return prisma.session.create({
+    data: session,
+  });
+}
+
+async function listSessions(prisma: PrismaClient, userId: string) {
+  return prisma.session.findMany({
+    where: {
+      createdBy: userId,
+    },
+  });
+}
+
+async function revoke(prisma: PrismaClient, sessionId: string, userId: string) {
+  try {
+    return await prisma.session.update({
+      where: {
+        createdBy_id: {
+          id: sessionId,
+          createdBy: userId,
+        },
+      },
+      data: {
+        revokedAt: new Date(),
+      },
+    });
+  } catch (e) {
+    throw new ApolloError('Failed to update session', 'SESSION_UPDATE_FAILED');
+  }
+}
+
+export async function generateTokenAndSession(
```

Comment:
If it's exported as part of the session model (available via prisma.session.generateTokenAndSession) let's not export it again and access it separately. Same applies to other functions in every model. There is just one export and all those functions should be only accessible through prisma

Reply:
fixed as part of https://github.com/makerdao-ses/switchboard-boilerplate/pull/28#discussion_r1151494134